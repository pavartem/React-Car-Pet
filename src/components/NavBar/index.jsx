import React from 'react';

const NavBar = ({theme}) => {
    return (
        <nav className={`navbar navbar-${theme} bg-${theme}`}>
            <div className="navbar-brand">
                Redux Hooks
            </div>
        </nav>
    );
};

export default NavBar;
