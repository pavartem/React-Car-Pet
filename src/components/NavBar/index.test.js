import React from "react";
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import NavBar from "./index";

configure({adapter: new Adapter()});

describe('<NavBar/>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<NavBar/>);
    })

    it('Should render component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render right classes', () => {
        wrapper.setProps({theme: 'dark'});
        expect(wrapper.contains(<nav className={`navbar navbar-dark bg-dark`}>
            <div className="navbar-brand">
                Redux Hooks
            </div>
        </nav>)).toBeTruthy();
    });

    it('Should render right text', () => {
        expect(wrapper.text()).toEqual('Redux Hooks');
    });
});
