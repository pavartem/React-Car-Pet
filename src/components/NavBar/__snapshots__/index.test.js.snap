// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> Should render component 1`] = `ShallowWrapper {}`;
